crates/blockchain-tree/src/block_buffer.rs

```diff
@@ -1,42 +1,38 @@
+use crate::metrics::BlockBufferMetrics;
 use lru::LruCache;
-use reth_primitives::{BlockHash, BlockNumHash, BlockNumber, SealedBlockWithSenders};
+use reth_primitives::{BlockHash, BlockNumber, SealedBlockWithSenders};
 use std::{
-    collections::{btree_map::Entry, hash_map, BTreeMap, HashMap, HashSet},
+    collections::{btree_map, hash_map, BTreeMap, HashMap, HashSet},
     num::NonZeroUsize,
 };
 
-use crate::metrics::BlockBufferMetrics;
-/// Type that contains blocks by number and hash.
-pub type BufferedBlocks = BTreeMap<BlockNumber, HashMap<BlockHash, SealedBlockWithSenders>>;
-
-/// Contains the Tree of pending blocks that are not executed but buffered
-/// It allows us to store unconnected blocks for potential inclusion.
+/// Contains the tree of pending blocks that cannot be executed due to missing parent.
+/// It allows to store unconnected blocks for potential future inclusion.
 ///
-/// It has three main functionality:
+/// The buffer has three main functionalities:
 /// * [BlockBuffer::insert_block] for inserting blocks inside the buffer.
 /// * [BlockBuffer::remove_with_children] for connecting blocks if the parent gets received and
 ///   inserted.
-/// * [BlockBuffer::clean_old_blocks] to clear old blocks that are below finalized line.
+/// * [BlockBuffer::remove_old_blocks] to remove old blocks that precede the finalized number.
 ///
 /// Note: Buffer is limited by number of blocks that it can contain and eviction of the block
 /// is done by last recently used block.
 #[derive(Debug)]
 pub struct BlockBuffer {
-    /// Blocks ordered by block number inside the BTreeMap.
-    ///
-    /// Note: BTreeMap is used so that we can remove the finalized old blocks
-    /// from the buffer
-    pub(crate) blocks: BufferedBlocks,
-    /// Needed for removal of the blocks. and to connect the potential unconnected block
-    /// to the connected one.
-    pub(crate) parent_to_child: HashMap<BlockHash, HashSet<BlockNumHash>>,
-    /// Helper map for fetching the block num from the block hash.
-    pub(crate) hash_to_num: HashMap<BlockHash, BlockNumber>,
+    /// All blocks in the buffer stored by their block hash.
+    pub(crate) blocks: HashMap<BlockHash, SealedBlockWithSenders>,
+    /// Map of any parent block hash (even the ones not currently in the buffer)
```

Comment:
why would we track block hashes of blocks that are not in the buffer?

Reply:
because we want the ability to lookup buffered blocks by their parent